src/test/scala/vsys/settings/WebhookEventRulesSpec.scala

```diff
@@ -41,16 +46,75 @@ class WebhookEventRulesSpec extends FlatSpec with Matchers {
+    when(mockTx2.transaction).thenReturn(tx2)
+
+    AfterHeight(10).applyRule(10, blockTime1, mockTx1, Set.empty) shouldBe(true)
+    AfterHeight(10).applyRule(3, blockTime1, mockTx1, Set.empty) shouldBe(false)
+
+    AfterTime(10).applyRule(0, blockTime1, mockTx1, Set.empty) shouldBe(true)
+    AfterTime(10).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(false)
+
+    WithTxs(true).applyRule(0, blockTime1, mockTx1, Set.empty) shouldBe(true)
+
+    WithMintingTxs(true).applyRule(0, blockTime1, mockTx1, Set.empty) shouldBe(true)
+    WithMintingTxs(true).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(true)
+    WithMintingTxs(false).applyRule(0, blockTime1, mockTx1, Set.empty) shouldBe(false)
+
+    val addr1 = mock[Address]
+    val addr2 = mock[Address]
+    when(addr1.toString).thenReturn("addr1")
+    when(addr2.toString).thenReturn("addr2")
+    RelatedAccs(Seq("addr1")).applyRule(0, blockTime1, mockTx1, Set(addr1, addr2)) shouldBe(true)
+    RelatedAccs(Seq("addr1")).applyRule(0, blockTime1, mockTx1, Set(addr2)) shouldBe(false)
+    RelatedAccs(Seq("addr1")).applyRule(0, blockTime1, mockTx1, Set.empty) shouldBe(false)
+    RelatedAccs(Seq.empty).applyRule(0, blockTime1, mockTx1, Set.empty) shouldBe(true)
+
+    IncludeTypes(Seq(1)).applyRule(0, blockTime1, mockTx1, Set.empty) shouldBe(false)
+    IncludeTypes(Seq(1, 2)).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(true)
+    IncludeTypes(Seq.empty).applyRule(0, blockTime1, mockTx1, Set.empty) shouldBe(false)
+
+    ExcludeTypes(Seq(1)).applyRule(0, blockTime1, mockTx1, Set.empty) shouldBe(true)
+    ExcludeTypes(Seq(1, 2)).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(false)
+    ExcludeTypes(Seq.empty).applyRule(0, blockTime1, mockTx1, Set.empty) shouldBe(true)
+
+    Amount(250L, 0, 1000, 1000, true).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(false)
+    Amount(249L, 0, 1000, 1000, true).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(true)
+    Amount(150L, 0, 1000, 1000, false).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(false)
+    Amount(149L, 0, 1000, 1000, false).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(true)
+
+    Amount(0, 150L, 1000, 1000, false).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(true)
+    Amount(0, 151L, 1000, 1000, false).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(false)
+    Amount(0, 250L, 1000, 1000, true).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(true)
+    Amount(0, 251L, 1000, 1000, true).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(false)
+
+    Amount(0, 0, 151, 1000, false).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(true)
+    Amount(0, 0, 150, 1000, false).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(false)
+
+    Amount(0, 0, 1000, 150, false).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(true)
+    Amount(0, 0, 1000, 149, false).applyRule(0, blockTime2, mockTx2, Set.empty) shouldBe(false)
+  }
```

Comment:
`Amount(gtVal, gteVal, ltVal, lteVal, withFeeVal)`. It is important to test all boundary cases. These will help you find the bug in the code.

For example:
if you amount is x, you should test, `Amount(x/x-1, _, _, _, _)` and `Amount(_, x/x+1, _, _, _)`